generate/generate.go

```diff
@@ -1419,25 +1419,8 @@ func (g *Generator) DropProcessCapabilityPermitted(c string) error {
 	return validate.CapValid(cp, false)
 }
 
-func mapStrToNamespace(ns string, path string) (rspec.LinuxNamespace, error) {
-	switch ns {
-	case "network":
-		return rspec.LinuxNamespace{Type: rspec.NetworkNamespace, Path: path}, nil
-	case "pid":
-		return rspec.LinuxNamespace{Type: rspec.PIDNamespace, Path: path}, nil
-	case "mount":
-		return rspec.LinuxNamespace{Type: rspec.MountNamespace, Path: path}, nil
-	case "ipc":
-		return rspec.LinuxNamespace{Type: rspec.IPCNamespace, Path: path}, nil
-	case "uts":
-		return rspec.LinuxNamespace{Type: rspec.UTSNamespace, Path: path}, nil
-	case "user":
-		return rspec.LinuxNamespace{Type: rspec.UserNamespace, Path: path}, nil
-	case "cgroup":
-		return rspec.LinuxNamespace{Type: rspec.CgroupNamespace, Path: path}, nil
-	default:
-		return rspec.LinuxNamespace{}, fmt.Errorf("unrecognized namespace %q", ns)
-	}
+func mapStrToNamespace(ns rspec.LinuxNamespaceType, path string) (rspec.LinuxNamespace, error) {
+	return rspec.LinuxNamespace{Type: ns, Path: path}, nil
```

Comment:
SGTM. However,  if you change this way, you can't judge whether the added type value is correct.

Reply:
Suggestions on a better way?